numpy/lib/function_base.py

```diff
@@ -1801,6 +1801,10 @@ def place(arr, mask, vals):
            [44, 55, 44]])
 
     """
+    if not isinstance(arr, np.ndarray):
```

Comment:
Could we use the same duck-typing solution as above, with the try/except clause?